lib/fluent/plugin/in_s3.rb

```diff
@@ -202,6 +216,19 @@ def setup_credentials
           credentials_options[:client] = Aws::STS::Client.new(:region => @s3_region)
         end
         options[:credentials] = Aws::AssumeRoleCredentials.new(credentials_options)
+      when @web_identity_credentials
+        c = @web_identity_credentials
+        credentials_options[:role_arn] = c.role_arn
+        credentials_options[:role_session_name] = c.role_session_name
+        credentials_options[:web_identity_token_file] = c.web_identity_token_file
+        credentials_options[:policy] = c.policy if c.policy
+        # TODO: Validation or clamping? If we go above 12 hours, the request will fail...
+        #       Is it better behavior to hard fail, or just provide the max session duration?
```

Comment:
How about putting the warning? I think that validation about AWS parameter should be done by AWS, fluentd shouldn't have the knowledge about it.

Reply:
Sorry, I'm pretty new to ruby and fluentd -- are you talking about ruby's standard [warning lib](https://ruby-doc.org/core-2.5.0/Warning.html)?

That said, I think you're right. It definitely makes more sense to let AWS validate the parameters. If they ever change the bounds, then we'd have to update the validation (including a warning). I'd personally lean towards not adding the warning in code, and just leaving it in the docs. Happy to add it though if you feel strongly about it!

Reply:
Oh, I see now you're talking about `log.warn` right?